### ISSUE TITLE
`call_nth/2` cutting

### ISSUE THREAD

**Issue by notoria** [trealla-prolog/trealla#626]
```
$ ./tpl -v
Trealla Prolog (c) Infradig 2020-2022, v1.34.20
$ ./tpl
?- [user].
q :- repeat, call_nth(true, 1).
   true.
?- q.
   true. % Expected choice point.
?-
```

**Comment by infradig** [comment]
$ tpl
?- [user]. q :- repeat, call_nth(true, 1).
   true.
?- q.
   true
;  true
;  ... .

**Comment by notoria** [comment]
Indeed, solved in `v1.34.21`.
